airflow/providers/amazon/aws/hooks/redshift.py

```diff
@@ -126,3 +135,101 @@ def create_cluster_snapshot(self, snapshot_identifier: str, cluster_identifier:
             ClusterIdentifier=cluster_identifier,
         )
         return response['Snapshot'] if response['Snapshot'] else None
+
+
+class RedshiftSQLHook(DbApiHook):
+    """
+    Execute statements against Amazon Redshift, using redshift_connector
+
+    This hook requires the redshift_conn_id connection. This connection must
+    be initialized with the host, port, login, password. Additional connection
```

Comment:
If the connection requires `host`, `port`, `login` and `password`, it would be a good idea if there was logic in the `_get_conn_params()` method to verify and raise an exception if any are missing.

Reply:
Is `schema` (aka `database` in this connection's case) required as well?

Reply:
yeah i think this may just be the wrong language:

> This connection must be initialized with the host, port, login, password

i think they are all optional depending on the circumstance.  e.g. if you pass `cluster_identifier` as a kwarg (through extra) i don't think you would need host.  and if you are using iam you would not need password. and you don't need port, this i've verified.

Reply:
> i think they are all optional depending on the circumstance. 
this is correct, thank you both for identifying this mistake

Reply:
fixed in eac5cdacd75c0783f3418af24185f6e009a49c06, only schema is stated as required